packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/SimulationController.java

```diff
@@ -345,7 +345,9 @@ public ResponseEntity<Dataset> createFromSimulationResult(
 		@PathVariable("id") final UUID id,
 		@PathVariable("project-id") final UUID projectId,
 		@RequestParam("dataset-name") final String datasetName,
-		@RequestParam("add-to-project") final Boolean addToProject
+		@RequestParam("add-to-project") final Boolean addToProject,
+		@RequestParam("model-configuration-id") final Optional<UUID> modelConfigurationId,
```

Comment:
I don't think `Optional` is right here, you can just use `UUID`. If you want to make these params optional it would be done like:
```
	@RequestParam(
			value = "model-configuration-id",
			required = false
		) final String modelConfigurationId,
```